cmd/keeper/keeper.go

```diff
@@ -842,6 +842,80 @@ func (p *PostgresKeeper) updateReplSlots(dbLocalState *DBLocalState, followersUI
 	return nil
 }
 
+func (p *PostgresKeeper) recreateAddonReplicationSlots(addonSlotNames []string) error {
+	var existingReplicationSlots []string
+	existingReplicationSlots, err := p.pgm.GetReplicationSlots()
```

Comment:
avoid retrieving them two times, just get them one time and pass it to this function and to `updateReplSlots`.